From 1a7deac0c036a773702b5fb00038fa7301db759c Mon Sep 17 00:00:00 2001
From: "Subhransu S. Prusty" <subhransu.s.prusty@intel.com>
Date: Fri, 23 May 2014 17:27:45 +0530
Subject: ASoC: intel: Enable Internal Mic

Micbias is required to record through Internal Mic. Add it to DAPM path.
Also Headset Mic is connected to IN2 and Internal mic to IN1. Fix the
routing in machine driver.

Issue: GMIN-294
Change-Id: I20166a54f4a93e8c8dea3e53c4c8677f515a06bd
Signed-off-by: Subhransu S. Prusty <subhransu.s.prusty@intel.com>
---
 sound/soc/intel/board/byt_cr_dpcm_rt5640.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/sound/soc/intel/board/byt_cr_dpcm_rt5640.c b/sound/soc/intel/board/byt_cr_dpcm_rt5640.c
index 153320f..7178137 100644
--- a/sound/soc/intel/board/byt_cr_dpcm_rt5640.c
+++ b/sound/soc/intel/board/byt_cr_dpcm_rt5640.c
@@ -559,9 +559,12 @@ static const struct snd_soc_dapm_widget byt_dapm_widgets[] = {
 };
 
 static const struct snd_soc_dapm_route byt_audio_map[] = {
-	{"IN1P", NULL, "Headset Mic"},
-	{"IN1N", NULL, "Headset Mic"},
-	{"IN2P", NULL, "Int Mic"},
+	{"IN2P", NULL, "Headset Mic"},
+	{"IN2N", NULL, "Headset Mic"},
+	{"IN1P", NULL, "micbias1"},
+	{"LDO2", NULL, "Int Mic"},
+	{"micbias1", NULL, "Int Mic"},
+
 	{"Headphone", NULL, "HPOL"},
 	{"Headphone", NULL, "HPOR"},
 	{"Ext Spk", NULL, "SPOLP"},
-- 
1.9.3

